<launch>
	<arg name="model" default="huboplus.urdf" />

	<param name="robot_description" textfile="$(find hubo_urdf)/$(arg model)" />
	<group ns="hubo/hardware">
		<node name="hw_state_publisher" pkg="hubo_motion_ros" type="hubo_hw_state_publisher" />
	</group>
	<node name="robot_tf_publisher" pkg="robot_state_publisher" type="state_publisher" >
		<remap from="/joint_states" to="/hubo/joint_states"/>
	</node>
	
	<group ns="hubo/vision">
		<node name="recognise_table_objects" pkg="hubo_drc_vision" type="recognise_table_objects" />
	</group>
	
	<group ns="hubo/planning">
		<node name="pseudo_grasp_planner" pkg="hubo_drc_vision" type="pseudo_grasp_planner" />
	</group>
	
	<group ns="hubo/motion">
		<node name="hubo_manip_traj_forwarder" pkg="hubo_motion_ros" type="hubo_manip_traj_forwarder" />
	</group>
</launch>
